test/test_view_dicom_tree.py

```diff
@@ -0,0 +1,158 @@
+import os
+import pytest
+import platform
+
+from src.Controller.GUIController import MainWindow
+from src.Model.PatientDictContainer import PatientDictContainer
+from src.View.ImageLoader import ImageLoading
+from src.Model.GetPatientInfo import DicomTree
+
+from pydicom import dcmread
+from pydicom.errors import InvalidDicomError
+
+
+def get_dicom_files(directory):
+    """
+    Function to find DICOM files in a given folder.
+    :param directory: File path of folder to search.
+    :return: List of file paths of DICOM files in given folder.
+    """
+    dicom_files = []
+
+    # Walk through directory
+    for root, dirs, files in os.walk(directory, topdown=True):
+        for name in files:
+            # Attempt to open file as a DICOM file
+            try:
+                dcmread(os.path.join(root, name))
+            except (InvalidDicomError, FileNotFoundError):
+                pass
+            else:
+                dicom_files.append(os.path.join(root, name))
+    return dicom_files
+
+
+def recursive_search(dict_tree, parent):
+    """
+    Recursive Function to test all rows match the data from the dictionary
+    :param dict_tree: The dictionary to be compared to
+    :param parent: Parent node of the DICOM Tree
+    """
+    count = 0  # Keep track of rows
+    for key in dict_tree:
+        value = dict_tree[key]  # get value from dict tree
+        if isinstance(value, type(dict_tree)):  # if dict_tree object in row
+            child = parent.child(count)
+            recursive_search(value, child)
+        else:
+            # Check row matches
+            assert parent.child(count, 0).text() == key
+            assert parent.child(count, 1).text() == str(value[0])
+            assert parent.child(count, 2).text() == str(value[1])
+            assert parent.child(count, 3).text() == str(value[2])
+            assert parent.child(count, 4).text() == str(value[3])
+        count += 1
+    return count
+
+
+class TestDICOMTreeTab:
+    """Class to set up the OnkoDICOM main window for testing the structures tab."""
+    __test__ = False
+
+    def __init__(self):
+        # Load test DICOM files and set path variable
+        path = "/testdata/DICOM-RT-TEST"
```

Comment:
This anti pattern has become prevalent.
Use pathlib to avoid coding to specific platforms.

Reply:
I agree